File: 2024/05/p1.test.lua
local describe = require("lib.test")
local p1 = require(".2024.05.p1")

describe("input", function(it)
  it("example", function()
  local example = [[47|53
97|13
97|61
97|47
75|29
61|13
75|53
29|13
97|29
53|29
61|53
97|53
61|29
47|13
75|47
97|75
47|61
75|61
47|29
75|13
53|13

75,47,61,53,29
97,61,53,29,13
75,29,13
75,97,47,61,53
61,13,29
97,13,75,29,47]]
    local expected = 143

    local n = p1.solve(example)

    assert(n == expected, string.format("Expected %d, but was %d", expected, n))
  end)

  it("input", function()
    local fp = io.open("./2024/05/assets/input.txt", "r") or error("Input not found")
    local input = fp:read("*a")
    local expected = 4872

    local n = p1.solve(input)

    assert(n == expected, string.format("Expected %d, but was %d", expected, n))
  end)
end)

File: 2024/13/__tests__/p1.test.ts
import assert from "node:assert";
import { readFile } from "node:fs/promises";
import { describe, it } from "node:test";
import { solve } from "../p1.js";
import path from "node:path";

describe("p1", () => {
  it("example 1", () => {
    const example = `Button A: X+94, Y+34
Button B: X+22, Y+67
Prize: X=8400, Y=5400`

    assert.strictEqual(solve(example), 480);
  });

  it("input", async () => {
    const filename = path.join(import.meta.dirname, "../assets/input.txt").replace("/dist/", "/");
    const input = await readFile(filename, "utf8");

    assert.strictEqual(solve(input), 28753);
  });
});